File: Analysis/bin/plot.cc
#include <string>
#include <vector>

#include <ChargedAnalysis/Utility/include/parser.h>
#include <ChargedAnalysis/Analysis/include/plotter1D.h>
#include <ChargedAnalysis/Analysis/include/plotter2D.h>

int main(int argc, char *argv[]){
    //Parser arguments
    Parser parser(argc, argv);

    std::string histDir =  parser.GetValue<std::string>("hist-dir");
    std::string channel = parser.GetValue<std::string>("channel");
    std::vector<std::string> processes = parser.GetVector<std::string>("processes");
    std::vector<std::string> outDirs = parser.GetVector<std::string>("out-dirs");

    //Call and run Plotter1D class
    Plotter1D plotter1D(histDir, channel, processes);
    plotter1D.ConfigureHists();
    plotter1D.Draw(outDirs);
}

File: Analysis/python/plot.py
from task import Task

import os

class Plot(Task):
    def __init__(self, config = {}):
        super().__init__(config)

    def run(self):
        self["executable"] = "plot"

        self["arguments"] = [
                "--hist-dir", self["hist-dir"], 
                "--channel", self["channel"], 
                "--processes", *self["processes"], 
                "--out-dirs", self["dir"], self["web-dir"],
                "--era", self["era"]
        ]

    def output(self):
        self["output"] = ["{}/{}.pdf".format(self["dir"], "kappa")]

    @staticmethod
    def configure(config, haddTasks, channel, era):        
        task = {
                "name": "Plot_{}_{}".format(channel, era), 
                "channel": channel, 
                "hist-dir": os.environ["CHDIR"] + "/{}".format(config["dir"].replace("[C]", channel).replace("[E]", era)), 
                "dir":  os.environ["CHDIR"] + "/Plots/{}".format(config["dir"].replace("[C]", channel).replace("[E]", era)), 
                "web-dir": os.environ["CHDIR"] + "/CernWebpage/Plots/{}".format(config["dir"].replace("[C]", channel).replace("[E]", era)), 
                "display-name": "Plots: {}/{}".format(channel, era), 
                "dependencies": [t["name"] for t in haddTasks if t["channel"] == channel], 
                "processes": list(set([t["process"] for t in haddTasks if t["channel"] == channel])),
                "era": era
        }

        return [Plot(task)]
